elementor/elementor
# Fix: WP CLI - Only 100 sites are processed when running on a multisite [ED-7834] (Fixes #18639)

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* While running CLI commands in a multisite environment, process all sites instead of only the first 100

## Description
This fix addresses issue #18639 where all multisite-related commands only loop over the first 100 sites, due to the default value of the `number` argument for the `WP_Site_Query` used by the `get_sites` function.

Since we only ever use the `blog_id`, I've limited the results to being an array of `blog_id`s as well, so we don't clog up the memory with `WP_Site` objects that remain unused.


## Test instructions
This PR can be tested by following these steps:

* Have a multisite with more than 100 blogs in it.
* Run any of the commands that call `get_sites`
* See that all sites are now getting processed instead of just the first 100.

## Quality assurance


Fixes #18639
